R/value.R

```diff
@@ -180,24 +180,24 @@ inspect_lt_value <- function(
     return(failure("Passed comparator value is NULL"))
   }
 
-  if (!is.numeric(value)) {
-    return(failure("Passed value is not a numerical"))
-  }
+  res <- inspect_real_value(
+    value,
+    allow_na = allow_na,
+    allow_null = allow_null
+  )
 
-  if (!is.numeric(comparator)) {
-    return(failure("Passed comparator value is not a numerical"))
+  if (!res$valid) {
+    return(res)
   }
 
-  if (length(value) != 1) {
-    return(failure(
-      "Passed value must be a single numerical value, not a vector"
-    ))
-  }
+  res <- inspect_real_value(
+    comparator,
+    allow_na = allow_na,
+    allow_null = allow_null
```

Comment:
il comparator non puo' mai essere null

Reply:
two questions:
1. comparator can never be NA either, right?
2. there exist already a check on the comparator being NULL at line 179
3. if I remove both lines 195 and 196, then inspect_real_value will use the default for both allow_na and allow_null, which is FALSE in both cases, right?